File: server/rest/src/main/kotlin/senegai/server/restapi/wto/SilvaOptionumSearchResultWTO.kt
/* @tt{{{

    @move-comment-backward
    @template-renderer [
        templateRendererClassName="ItemSearchResultWTORenderer"
        templateRendererPackageName="senegai.codegen.renderer.be"
        templateRendererInterfaceName="BeItemRenderer"
        templateRendererInterfacePackageName="senegai.codegen.renderer.be"
    ] [
        modelClassName="BeItemModel"
        modelPackageName="senegai.codegen.renderer.model.be"
        modelName="model"
    ]

    @replace-value-by-expression
        [ searchValue="SilvaOptionum" replaceByExpression="model.itemName.pascalCase" ]
        [ searchValue="silvaOptionum" replaceByExpression="model.itemName.camelCase" ]

    @modify-provided-filepath-by-replacements

}}}@ */
package senegai.server.restapi.wto

data class SilvaOptionumSearchResultWTO(
    val silvaOptionumList: List<SilvaOptionumWTO>,
)

File: client/src/app/wto/silva-optionum-search-result.wto.ts
/* @tt{{{

    @move-comment-backward
    @template-renderer [
        templateRendererClassName="ItemSearchResultWtoRenderer"
        templateRendererPackageName="senegai.codegen.renderer.angular"
        templateRendererInterfaceName="UiItemRenderer"
        templateRendererInterfacePackageName="senegai.codegen.renderer.angular"
    ] [
        modelClassName="UiItemModel"
        modelPackageName="senegai.codegen.renderer.model.ui"
        modelName="model"
    ]

    @replace-value-by-expression
        [ searchValue="SilvaOptionum" replaceByExpression="model.itemName.pascalCase" ]
        [ searchValue="silvaOptionum" replaceByExpression="model.itemName.camelCase" ]
        [ searchValue="silva-optionum" replaceByExpression="model.itemName.kebabCase" ]

    @modify-provided-filepath-by-replacements

}}}@ */
import {SilvaOptionumWTO} from "@app/wto/silva-optionum.wto";

/**
 * The Silva Optionum search result WTO (Web Transfer Object), returned by the search
 * endpoint. It wraps the found Silva Optionums.
 */
export interface SilvaOptionumSearchResultWTO {
    silvaOptionumList: Array<SilvaOptionumWTO>;
}
